deploy/olm-catalog/observatorium-operator/0.0.1/observatorium-operator.v0.0.1.clusterserviceversion.yaml

```diff
@@ -0,0 +1,114 @@
+apiVersion: operators.coreos.com/v1alpha1
```

Comment:
This file (and all of the files in /deploy/olm-catalog) are not actually deployed to the cluster, they are only files used to build the docker container. I think they should _not_ live in /deploy because that will cause confusion around deploying OLM
How are these files generated? If they are the output of a generator binary, then that binary should simply be run in the dockerfile

Reply:
Right, the files in /deploy/olm-catalog are not deployed to the cluster, only those in /deploy/olm are.
These files were generated using the operator-sdk on a separate project as we were not sure this project follows the structure expected by the operator-sdk (as far as we understand, this operator was not generated by the operator-sdk). We placed those files here only because we saw this convention [elsewhere](https://github.com/openshift-kni/performance-addon-operators/tree/master/deploy/olm-catalog) but this is just a temporary state as the next step would be to try generating these files here - and doing that inside the container is a nice idea.

Reply:
great, let's do that then, let's generate everything in a multi-stage container so we can leave this clean :)

Reply:
Are we still working on removing this from the repo?

Reply:
This is indeed not pretty but apparently we cannot avoid this until image indexing is supported by the OLM catalog. It goes like that:
* Until OpenShift 4.5 where image indexing will be supported, we have to store the history of the versions for upgrades flow. For instance, let's say we are going to release 0.0.3 which will be set to replace 0.0.2, then we'll have to deploy 0.0.1 (that is replaced by 0.0.2) and 0.0.2 as well  - without storing them in the repo, we won't have access to their CSV and CRD resources. So the olm-catalog folder needs to remain, we can move it elsewhere if needed though (that's the default place the operator-sdk uses as well as in KubeVirt and Performance-Addon-Operator).
* As from OpenShift 4.5, when image indexing is supported, we won't need each release to provide all those resources for the upgrade chain.

So we containerized the generation of the CSV, that pulls the operator-sdk, builds the utilities and generates the resources. That container is used by a dedicated target within the Makefile for now, later (with image indexing) we can unify the Dockerfiles Dockerfile.csv and Dockerfile.olm into a multi-stage container.

@squat  Would that make sense?

Reply:
hmm no, I don't understand that. Why don't we simply build the CSV and bake it into the OLM Docker image (rather than a CSV Docker image) without it ever touching the repo?

Reply:
We can do that now, when we're releasing the first OLM based version, let's call it 0.0.1. We'll face an issue when we'll reach the point we release the next version - the next version, let's say 0.0.2, would be set to replace 0.0.1, right? without placing the CRD and CSV of 0.0.1 in the registry container, users will not be able to deploy 0.0.2 (OLM currently assumes the registry container hold the information that is needed for the whole upgrade chain). So if we don't place it on the repo, how would we have access to the CRD and CSV of 0.0.1, when we'll reach 0.0.2?